compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -634,7 +634,7 @@ pub type TyAndLayout<'tcx> = rustc_target::abi::TyAndLayout<'tcx, Ty<'tcx>>;
 pub trait LayoutOfHelpers<'tcx>: HasDataLayout + HasTyCtxt<'tcx> + HasParamEnv<'tcx> {
     /// The `TyAndLayout`-wrapping type (or `TyAndLayout` itself), which will be
     /// returned from `layout_of` (see also `handle_layout_err`).
-    type LayoutOfResult: MaybeResult<TyAndLayout<'tcx>>;
+    type LayoutOfResult: MaybeResult<TyAndLayout<'tcx>> = TyAndLayout<'tcx>;
```

Comment:
Is there any case where this is not `TyAndLayout<'tcx>` and `FnAbiResult` is not `&'tcx FnAbi<'tcx, Ty<'tcx>>`?

Reply:
For both cases there are several places where it's `Result<TyAndLayout<'tcx>, _>` and `Result<&'tcx FnAbi<'tcx, Ty<'tcx>>, _>`.